ecosystem/typescript/sdk/src/aptos_types/transaction.ts

```diff
@@ -371,25 +371,61 @@ export abstract class RawTransactionWithData {
     switch (index) {
       case 0:
         return MultiAgentRawTransaction.load(deserializer);
+      case 1:
+        return MultiAgentRawTransaction.load_fee_payer(deserializer);
       default:
         throw new Error(`Unknown variant index for RawTransactionWithData: ${index}`);
     }
   }
 }
 
 export class MultiAgentRawTransaction extends RawTransactionWithData {
+  private fee_payer_address?: AccountAddress;
+
   constructor(
     public readonly raw_txn: RawTransaction,
     public readonly secondary_signer_addresses: Seq<AccountAddress>,
   ) {
     super();
   }
 
+  set_fee_payer_address(fee_payer_address: AccountAddress): void {
+    this.fee_payer_address = fee_payer_address;
+  }
+
+  get_authenticator(
+    sender: AccountAuthenticator,
+    secondary_signers: Seq<AccountAuthenticator>,
+    fee_payer?: AccountAuthenticator,
+  ): TransactionAuthenticatorMultiAgent {
+    if (secondary_signers.length !== this.secondary_signer_addresses.length) {
+      throw new Error("Number of secondary signers does not match.");
+    }
+    let fee_pay;
+    if (this.fee_payer_address) {
+      if (!fee_payer) {
+        throw new Error("Fee payer is not set.");
+      }
+      fee_pay = { address: this.fee_payer_address, authenticator: fee_payer };
+    } else if (fee_payer) {
+        throw new Error("Fee payer is set.");
```

Comment:
I am a bit confused about the implementation here? looks like if we dont have `this.fee_payer_address` and/or `fee_payer` we want to throw an error? and if we have both, we want to use it as the `fee_pay`?
could we make it more clear?

Reply:
yes a tx with a fee payer must be signed with a fee payer, and tx without a fee payer must not have a fee payer signature.

Reply:
can we add it as a comment?